File: 1967-number-of-strings-that-appear-as-substrings-in-word/1967-number-of-strings-that-appear-as-substrings-in-word.java
class Solution {
    public int numOfStrings(String[] patterns, String word) 
    {
        int count = 0;

        for(String s : patterns)
        {
            for(int i = 0; i <= (int)word.length() - (int)s.length(); i++)
            {
                if(word.substring(i, i + s.length()).equals(s))
                {
                    count++;
                    break;
                }
            }
        }
        return count;
    }
}

File: 1967-number-of-strings-that-appear-as-substrings-in-word/1967-number-of-strings-that-appear-as-substrings-in-word.cpp
class Solution {
public:
    int numOfStrings(vector<string>& patterns, string word) 
    {
        int count = 0;

        for(int i = 0;  i < patterns.size(); i++)
        {
            string p = patterns[i];
            bool found = false;

            for(int j = 0;  j<= (int)word.size() - (int)p.size(); j++)
            {   
                int k;
                for(k = 0; k < p.size(); k++)
                {
                    if(word[j+k] != p[k])
                        break;
                }
                            
                if(k == p.size())
                {
                    found = true;
                    break;
                }
            }
            if(found == true)
            {
                count++;
            }
        }
        return count;
    }

};